pkg/pgsql/server/fmessages/bind.go

```diff
@@ -0,0 +1,154 @@
+// The response is either BindComplete or ErrorResponse.
+//
+// Note
+// The choice between text and binary output is determined by the format codes given in Bind, regardless of the SQL
+// command involved. The BINARY attribute in cursor declarations is irrelevant when using extended query protocol.
+type BindMsg struct {
+	// The name of the destination portal (an empty string selects the unnamed portal).
+	DestPortalName string
+	// The name of the source prepared statement (an empty string selects the unnamed prepared statement).
+	PreparedStatementName string
+	// The parameter format codes. Each must presently be zero (text) or one (binary).
+	ParameterFormatCodes []int16
+	// Array of the values of the parameters, in the format indicated by the associated format code. n is the above length.
+	ParamVals []interface{}
+	// The result-column format codes. Each must presently be zero (text) or one (binary).
+	ResultColumnFormatCodes []int16
+}
+
+func ParseBindMsg(payload []byte) (BindMsg, error) {
+	b := bytes.NewBuffer(payload)
+	r := bufio.NewReader(b)
+	destPortalName, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	preparedStatement, err := getNextString(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	// The number of parameter format codes that follow (denoted C below).
+	// This can be zero to indicate that there are no parameters or that the parameters all use the default format (text);
+	// or one, in which case the specified format code is applied to all parameters; or it can equal the actual number
+	// of parameters.
+	parameterFormatCodeNumber, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+	parameterFormatCodes := make([]int16, parameterFormatCodeNumber)
+	for k := 0; k < int(parameterFormatCodeNumber); k++ {
+		p, err := getNextInt16(r)
+		if err != nil {
+			return BindMsg{}, err
+		}
+		parameterFormatCodes[k] = p
+	}
+	// The number of parameter values that follow (possibly zero). This must match the number of parameters needed by the query.
+	pCount, err := getNextInt16(r)
+	if err != nil {
+		return BindMsg{}, err
+	}
+
+	// Handling format codes: see resultColumnFormatCodesNumber property comment
+	forceTXT := false
+	forceBIN := false
+	if len(parameterFormatCodes) == 0 {
+		forceTXT = true
+	}
+	if len(parameterFormatCodes) == 1 {
+		switch parameterFormatCodes[0] {
+		case 0:
+			forceTXT = true
+		case 1:
+			forceBIN = true
+		default:
+			return BindMsg{}, errors.New("malformed bind message. Allowed format codes are 1 or 0")
+		}
+	}
+
+	if len(parameterFormatCodes) > 1 && len(parameterFormatCodes) != int(pCount) {
+		return BindMsg{}, errors.New("malformed bind message. Parameters format codes didn't match parameters count")
+	}
+
+	params := make([]interface{}, 0)
+	for i := 0; i < int(pCount); i++ {
+		pLen, err := getNextInt32(r)
+		if err != nil {
+			return BindMsg{}, err
+		}
+		pVal := make([]byte, pLen)
```

Comment:
`pLen` can be of any size including negative values (that would panic here) or very high ones (allocation of huge memory buffer), please sanitize that value